### PR TITLE
Capstone Project JavaScript

### PR DESCRIPTION
**Pull request by Nelitaa**
**Team  members:** @osamaashraf6 , @Nelitaa 

**Project requirements**

**API**
1. First, you need to find an API so you can base the development of the webapp around it. The API should allow you to:
- Get a list of items with a unique item id (or generate the unique id).
- For a given item, get detailed information about it.
2. You will use our [Involvement API](https://www.notion.so/869e60b5ad104603aa6db59e08150270) to record the different user interactions (likes, comments and reservations).

**Interfaces**
1. You should build these interfaces:
- The home page.
- The comments popup.
2. You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.

**Home page**
1. When the page loads, the webapp retrieves data from:
- The selected API and shows the list of items on screen.
- The Involvement API to show the item likes.
2. When the user clicks on the Like button of an item, the interaction is recorded in the Involvement API and the screen is updated.
3. When the user clicks on the "Comments" button, the Comments popup appears.
4. Home page header and navigation similar to the given mockup.
5. Home page footer similar to the given mockup.

**Comments popup**
1. When the popup loads, the webapp retrieves data from:
- The selected API and shows details about the selected item.
- The Involvement API to show the item comments.
2. When the user clicks on the "Comment" button, the data is recorded in the Involvement API and the screen is updated.

**Counters** 
1. We have counters in all the interfaces that show:
- The number of items (home).
- The number of comments (comments popup).
2. Even if the API gives you these numbers, you will create a specific function to calculate these numbers on each page. 
3. Respect the following rules:
- Each counter should be implemented as a separate module.
- A counter function should look for specific DOM elements (e.g. for the comments counter it should look for comments) and make the counting based on what is actually displayed on the page.
- A counter function should cover all the edge cases you can think about.
- These counter functions need to be covered with unit tests using Jest. Make sure that you have tests for all edge cases.

**Technical set up**
- Set up the repository on GitHub and use gitflow.
- Set up webpack.
- Use modules to organize your code. You should have a separate directory for your modules ( src/modules). It is your responsibility to divide your code into logical chunks.
- Set up a JavaScript testing library (Jest). All your tests should be stored in one directory.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,285 +1,63 @@
...
-
-### Usage
-
-To run the project, execute the following command:
+-Open up Github.
+-Clone this repository from GitHub by copying the HTTPS address. Write: git clone https://github.com/Nelitaa/Capstone-Project-JavaScript.git
+-On your VSCode hit "Open a new repository" and paste the URL.
+-Select the destination folder. Write: cd ECapstone-Project-JavaScript
+-All the files should be cloned to your local folder and are ready to be used.
+-Write: npm install
```
Comment (addod19) [inline README.md:32]: ### Optional
- [ ] It would be better if we add the instructions about how to run the test to the README file to make it more descriptive :man_technologist:

File: src/modules/test/render.test.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/**
+ * @jest-environment jsdom
+ */
+
```
Comment (addod19) [inline src/modules/test/render.test.js:4]: - [ ] It would be better if we can successfully run the test suite. However, I encountered what I think is a node modules problem as seen in the screenshot below
Screenshot from 2022-11-25 16-25-05

Here is what I did in trying to resolve it, I removed the node_modules and package-lock.json files. Did the `npm i` again then after `npm run test`. All these couldn't resolve it

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @Nelitaa and @osamaashraf6, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- [ ] Gitflow followed :rocket: 
- [ ] API functionality7 works as expected :100: 
- [ ] Interface looks great as expect, contains the homepage with the comments pop up actions :1st_place_medal: 
- [ ] Jacascript code is clean, followed single responsibility principle :100: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (Nelitaa) [reply README.md:46]: @addod19 Thank you very much for your review, we add the instruction for run the test :)

File: src/modules/test/render.test.js (hunk 2 above)
Comment (Nelitaa) [reply src/modules/test/render.test.js]: @addod19 Thank you very much for your review, now you can run **npm run test** and it will work 👯

**Review by DammyShittu — APPROVED** [review]
Hi @Nelitaa @osamaashraf6,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
